### ISSUE TITLE
Error in `Dataset.npred_signal`

### ISSUE THREAD

**Issue by AtreyeeS** [gammapy/gammapy#3490]
**Gammapy version**

present dev

**Bug description**

`MapDataset.npred_signal(model_name)` fails on the first call.
It works if `MapDataset.npred_signal()` (without arguments) is first called.

**To Reproduce**
 see this : https://github.com/AtreyeeS/test_notebooks/blob/master/error_npred.ipynb

Proposed solutions @registerrier @QRemy ?

**Comment by QRemy** [comment]
Thanks @AtreyeeS,
https://github.com/gammapy/gammapy/blob/86c341315e035ec3347a51afbbace5845d294d12/gammapy/datasets/map.py#L440 
these lines could be modified to filter the evaluators instead of returning directly  `self.evaluators[model_name].computed_npred()`, such as:
```python
evaluators = self.evaluators
 if model_name is not None:
    evaluators = {model.name: self.evaluators[model_name]} 
for evaluator in evaluators.values():
```

In  addition it could be interesting to have `model_names` as a list and add a `evaluators.select` so ones can easily get the stacked npred of multiple models using npred_signal.

**Comment by AtreyeeS** [comment]
Closed via #3500
